packages/api-client-core/src/InternalModelManager.ts

```diff
@@ -44,61 +45,78 @@ export const internalFindOneQuery = (apiIdentifier: string) => {
     `;
 };
 
-export const internalFindManyQuery = (apiIdentifier: string, isFirstQuery = false) => {
+export const internalFindFirstQuery = (apiIdentifier: string, options?: RecordData) => {
   const capitalizedApiIdentifier = capitalize(apiIdentifier);
-  const namePrefix = isFirstQuery ? "InternalFindFirst" : "InternalFindMany";
-  return `
-    query ${namePrefix}${capitalizedApiIdentifier}(
-      ${
-        isFirstQuery
-          ? ""
-          : `
-      $after: String
-      $before: String
-      $first: Int
-      $last: Int`
-      }
-      $search: String
-      $sort: [${capitalizedApiIdentifier}Sort!]
-      $filter: [${capitalizedApiIdentifier}Filter!]
-    ) {
-      ${internalHydrationPlan(apiIdentifier)}
-      internal {
-        list${capitalizedApiIdentifier}(
-          ${
-            isFirstQuery
-              ? `
-          first: 1`
-              : `
-          after: $after
-          before: $before
-          first: $first
-          last: $last
-          `
-          }    
-          search: $search
-          sort: $sort
-          filter: $filter
-        ) {
-          edges {
-            ${isFirstQuery ? "" : "cursor"}
-            node
-          }
-          ${
-            isFirstQuery
-              ? ""
-              : `
-          pageInfo {
-            hasNextPage
-            hasPreviousPage
-            startCursor
-            endCursor
-          }`
-          }
-        }
-      }
-    }
-    `;
+
+  const defaultVariables = {
+    ...(options?.search && { search: { value: options?.search, type: "String", required: false } }),
+    ...(options?.sort && { sort: { value: options?.sort, type: `${capitalizedApiIdentifier}Sort!`, list: true } }),
+    ...(options?.filter && { filter: { value: options?.filter, type: `${capitalizedApiIdentifier}Filter!`, list: true } }),
+  };
+
+  return query(
+    [
+      {
+        operation: "internal",
+        fields: [
+          {
+            operation: `list${capitalizedApiIdentifier}`,
+            fields: [
+              {
+                edges: ["node"],
+              },
+            ],
+            variables: {
+              ...defaultVariables,
+              first: { value: 1, type: "Int" },
```

Comment:
When I tried using this query builder (elsewhere), I was never able to hardcode values like here; it always seemed to use the value _only_ for type definition (so "number" here), but it was then list that field as accepting an external variable, which you'd have to set.

Can you confirm that this behaviour is _not_ what we'll get here?

Reply:
To elaborate: IMO, the desired behaviour for `findFirst` is that there is no variable for `first`; you get the hardcoded value of 1, and _that's it_. Otherwise, it's a misnomer, which I would find weird 😅

Reply:
Yeah, this is a strange thing about how `gql-query-builder` works, it returns both a query document AND query variables to then pass to that document. When you have values to pass to the document, why not just hardcode them in it? It totally could, but it also supports generating queries with variables you don't yet have the values for, and I guess they decided that all variables should be passed the same. This function returns a `plan` object that has both a `.query` and a `.variables` on it, and we pass those variables down to the execution bits later.

Reply:
@airhorns oh, I see what you're saying: we're just feeding those variables back in, like on line 238; and since we're not using `value: options?.first`, there's no way to override it externally, so it ends up being the same thing as if it was hardcoded.

I did a quick scan of the codebase, and the only other call to `internalFindManyQuery` is related to docs, so that's fine.

Reply:
> oh, I see what you're saying: we're just feeding those variables back in, like on line 238; and since we're not using value: options?.first, there's no way to override it externally, so it ends up being the same thing as if it was hardcoded.

That is correct. 

> I did a quick scan of the codebase, and the only other call to internalFindManyQuery is related to docs, so that's fine.

Yeah, I will update the `exampleInternalFindFirstQuery` to use the new method once this has been merged.